app/src/main/java/com/hhua/android/simpletodo/MainActivity.java

```diff
@@ -88,9 +93,13 @@ private void setupListViewListener(){
                 new AdapterView.OnItemLongClickListener() {
                     @Override
                     public boolean onItemLongClick(AdapterView<?> adapter, View item, int pos, long id) {
+                        // delete task from database
+                        Task task = items.get(pos);
+                        deleteItem(task);
+                        System.out.println("pos " + pos);
+                        System.out.println("size " + items.size());
                         items.remove(pos);
```

Comment:
One edge case: Add a new task, and then remove it. Task in database has been removed, however, the listview didn't update as expected.